src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessorTest.java

```diff
@@ -0,0 +1,132 @@
+import com.azure.storage.blob.models.BlobItemProperties;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.blobrouter.data.envelopes.Envelope;
+import uk.gov.hmcts.reform.blobrouter.data.envelopes.Status;
+import uk.gov.hmcts.reform.blobrouter.services.BlobReadinessChecker;
+import uk.gov.hmcts.reform.blobrouter.services.EnvelopeService;
+
+import java.time.OffsetDateTime;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static java.time.Instant.now;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@ExtendWith(MockitoExtension.class)
+class ContainerProcessorTest {
+
+    @Mock BlobServiceClient storageClient;
+    @Mock BlobProcessor blobProcessor;
+    @Mock BlobReadinessChecker blobReadinessChecker;
+    @Mock EnvelopeService envelopeService;
+
+    @Mock BlobContainerClient containerClient;
+    @Mock BlobClient blobClient;
+    @Mock PagedIterable<BlobItem> listBlobsResult;
+
+    ContainerProcessor containerProcessor;
+
+    @BeforeEach
+    void setUp() {
+        containerProcessor = new ContainerProcessor(
+            storageClient,
+            blobProcessor,
+            blobReadinessChecker,
+            envelopeService
+        );
+    }
+
+    @Test
+    void should_continue_processing_stale_envelope() {
+        // given
+        var envelope = new Envelope(
+            UUID.randomUUID(),
+            "some_container",
+            "hello.zip",
+            now().minus(ContainerProcessor.STALE_AGE).minusSeconds(1), // stale
+            null,
+            null,
+            Status.CREATED,
+            false
+        );
+
+        storageHasBlob(envelope.fileName, envelope.container);
+        dbHas(envelope);
+
+        // when
+        containerProcessor.process(envelope.container);
+
+        // then
+        verify(blobProcessor).continueProcessing(envelope.id, blobClient);
+        verifyNoMoreInteractions(blobProcessor);
+    }
+
+    @Test
+    void should_skip_envelope_that_is_not() {
+        // given
+        var envelope = new Envelope(
+            UUID.randomUUID(),
+            "some_container",
+            "hello.zip",
+            now().minus(ContainerProcessor.STALE_AGE).plusSeconds(10), // NOT stale
+            null,
+            null,
+            Status.CREATED,
+            false
+        );
+
+        storageHasBlob(envelope.fileName, envelope.container);
+        dbHas(envelope);
+
+        // when
+        containerProcessor.process(envelope.container);
+
+        // then
+        verifyNoInteractions(blobProcessor);
+    }
```

Comment:
Interestingly, unit test for `ContainerProcessor` did not exist until now.
Other tests (not related to continuing processing of envelopes in 'created' status) can be added on a separate PR.